lib/location/scan_cellular.c

```diff
@@ -162,6 +180,23 @@ void scan_cellular_execute(int32_t timeout, uint8_t cell_count)
 		goto end;
 	}
 
+	/* GCI searches are not done when in RRC connected mode. We are waiting for  */
+	LOG_DBG("%s", k_sem_count_get(&entered_rrc_idle) == 0 ?
+		"Waiting for the RRC connection release..." :
+		"RRC already in idle mode");
+
+	/* If semaphore is reset during the waiting period, the position request was canceled.*/
+	if (k_sem_take(&entered_rrc_idle, K_SECONDS(SCAN_CELLULAR_RRC_IDLE_WAIT_TIME)) == -EAGAIN) {
+		if (!running) {
+			goto end;
+		}
+		/* We're still running, i.e., the wait for RRC idle timed out */
+		LOG_WRN("RRC connection was not released in %d seconds.",
+			SCAN_CELLULAR_RRC_IDLE_WAIT_TIME);
+		goto end;
```

Comment:
We didn't get to RRC idle and we will use NCELLMEAS=1 results which are 10s old. I'm thinking if that's find...

@tokangas, @jhirsi

Reply:
I think we just have to accept that. Also, if the RRC connection is released, there's time difference between the measurements, might be even more than 10 seconds.

Reply:
If we go into GCI searches, only old data we have is for PCI neighbors. Everything else is fresh from the GCI search including current cell.